oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/introspection/SpringOpaqueTokenIntrospector.java

```diff
@@ -222,7 +227,17 @@ private OAuth2AuthenticatedPrincipal convertClaimsSet(Map<String, Object> claims
 			}
 			return v;
 		});
-		return new OAuth2IntrospectionAuthenticatedPrincipal(claims, authorities);
+		return this.authenticationConverter.convert(new OAuth2IntrospectionAuthenticatedPrincipal(claims, authorities));
```

Comment:
I think it would provide more value if the default authentication converter took the claim set and added the authorities instead of being a pass-through. The reason is that this allows the code to align with `JwtAuthenticationConverter` down the road.

In the case of `JwtAuthenticationConverter`, the instance moves from `Jwt` (essentially the claim set) to an `Authentication` (something with authorities), and I think that we should do the same thing here.

More concretely, I think the default converter should be invoked like this:

```java
return this.authenticationConverter.convert(() -> claims);
```

and the default converter should do the work of computing the granted authorities and constructing an `OAuth2IntrospectionAuthenticatedPrincipal`. Will that work for you?

Reply:
thanks @jzheaux for feedback. currently working on it. I was thinking to change `convertClaimsSet ` method return type to `Map<String, Object>` to return claims and then use `this.authenticationConverter.convert(() -> claims)` inside `introspect()` method instead of `convertClaimsSet()`. does this seem right ?

Reply:
Yes, I think that makes a lot of sense.